src/main/scala/com.snowplowanalytics.s3/loader/KinesisS3Emitter.scala

```diff
@@ -108,3 +126,65 @@ class KinesisS3Emitter(
   override def fail(records: java.util.List[EmitterInput]): Unit =
     s3Emitter.sendFailures(records)
 }
+
+object KinesisS3Emitter {
+
+  /** Type of row which determined according to schema of self describing data */
+  sealed trait RowType extends Product with Serializable {
+    def getName: String
+  }
+
+  object RowType {
+    /** Represents cases where row type could not be determined
+      * since either row is not valid json or it is not self
+      * describing json
+      */
+    case class PartitionError(errorDir: String) extends RowType {
+      override def getName: String = errorDir
+    }
+
+    /** Represents cases where type of row can be determined successfully
+      * e.g. does have proper schema key
+      */
+    case class SelfDescribing(rowType: String) extends RowType {
+      override def getName: String = rowType
+    }
+
+    case object UnexpectedError extends RowType {
```

Comment:
`UnexpectedError` is a fairly undesriptive name. Does it mean that we waiting for other errors? In our case this error happens only when record is not a valid string (?).

Reply:
I think you are right about name being undescriptive. `InvalidRecordString` is much more descriptive I think.